puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/nightly_packaging.groovy

```diff
@@ -0,0 +1,39 @@
+def setup_nightly_build_environment(args) {
+    def project_name = args.project_name
+    def git_url = args.git_url
+    def git_ref = args.git_ref
+    def ruby_version = args.ruby_version ?: env.ruby_version
+
+    dir(project_name) {
+        // the full checkout() is necessary for checking out specific commits,
+        // as opposed to just branches with git()
+        checkout([
+            $class : 'GitSCM',
+            branches : [[name: git_ref]],
+            extensions: [[$class: 'CleanCheckout']],
+            userRemoteConfigs: [[url: git_url]]
+        ])
+    }
+    dir('foreman-packaging') {
+        git(url: 'https://github.com/theforeman/foreman-packaging.git', branch: 'rpm/develop')
```

Comment:
I've noticed multiple jobs that have a git URL there will start to monitor it. For example, the installer is currently built on every merge to `rpm/develop` because the webhook is delivered. Would that also happen here?

Reply:
To which job are you referring? If a job has polling setup, then it will poll for changes on _every_ repo that is contained in the pipeline.

Reply:
Maybe it was a misdiagnosis from my side. The `test_develop` and `test_proxy_develop` jobs currently trigger a lot of builds. I don't exactly where I saw it so I don't mind merging and seeing where the pieces land ;)

Reply:
It looks like both [`test_develop`](http://ci.theforeman.org/job/test_develop/changes) and [`test_proxy_develop`](http://ci.theforeman.org/job/test_proxy_develop/changes) are both triggering properly?

Reply:
They are triggering, but the problem is that every part in the matrix triggers a RPM build job. On http://koji.katello.org/koji/packageinfo?packageID=996 you can see we build every git hash 4 or 5 times. @ehelms thought it was something with the interaction between matrix jobs and pipeline jobs.

Reply:
Oh oh! I am following you now. I believe that a pipeline rather than a matrix would solve this problem as long as there is only one stage that triggers the build job after all the test stages have completed.

Reply:
https://github.com/theforeman/foreman-infra/pull/890